.gitignore

```diff
@@ -0,0 +1,47 @@
+# Dependencies
+node_modules/
+frontend/node_modules/
+frontend/apps/*/node_modules/
+
+# Build outputs
+frontend/apps/*/dist/
+backend/build/
+backend/api/build/
+backend/.gradle/
+backend/api/.gradle/
+
+# Gradle wrapper (keep gradle wrapper jar and properties)
+!backend/gradle/wrapper/gradle-wrapper.jar
+!backend/gradle/wrapper/gradle-wrapper.properties
+
+# IDE files
+.idea/
+.vscode/
+*.swp
+*.swo
+
+# OS files
+.DS_Store
+Thumbs.db
+
+# Logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Environment variables
+.env
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# Test coverage
+coverage/
+
+# Temporary files
+tmp/
+temp/
+*.tmp
+*.temp
```

Comment:
I see all files don't have empty line in the end, for both backend and frontend. Isn't it a bad practice? Can it be validated and autofixed by some linters?